pgbench_wrapper/trigger_pgbench.py

```diff
@@ -0,0 +1,207 @@
+            elif re.search('latency', results[idx][0]):
+                results[idx][0] += "_ms"
+                results[idx][1] = self._num_convert(results[idx][1].split(" ", 1)[0])
+            elif re.search('processed', results[idx][0]):
+                try:
+                    results[idx][1] = self._num_convert(results[idx][1].split("/", 1)[0])
+                except AttributeError:
+                    pass
+        config.append(["timestamp", datetime.now()])
+        return {"config": config, "results": results, "raw_output_b64": raw_output_b64}
+
+    def _parse_stderr(self, stderr):
+        progress = []
+        for line in stderr.splitlines():
+            if "progress" in line:
+                progress.append({
+                    "timestamp": datetime.fromtimestamp(float(line.split(" ")[1])),
+                    "tps": float(line.split(" ")[3]),
+                    "latency_ms": float(line.split(" ")[6]),
+                    "stddev": float(line.split(" ")[9])
+                })
+        return progress
+
+    def _summarize_data(self, data, iteration, uuid, database, pgb_vers):
+        print("+{} PGBench Results {}+".format("-" * (50), "-" * (50)))
+        print("PGBench version: {}".format(pgb_vers))
+        print("")
+        print("UUID: {}".format(uuid))
+        print("Run: {}".format(iteration))
+        print("")
+        print("Database: {}".format(database))
+        print("")
+        print("PGBench run info:")
+        for line in data['config']:
+            print("          {}: {}".format(line[0], line[1]))
+        print("")
+        # I asked for a mai tai, and they brought me a pina colada,
+        # and I said no salt, NO salt on the margarita, but it had salt
+        # on it, big grains of salt, floating in the glass.
+        print("TPS report:")
+        for line in data['results']:
+            print("          {}: {}".format(line[0], line[1]))
+        print("")
+        print("+{}+".format("-" * (115)))
+
+    def emit_actions(self):
+        output = self._run_pgbench()
+        if output[2] == 1:
+            print("PGBench failed to execute, trying one more time..")
+            output = self._run_pgbench()
+            if output[2] == 1:
+                print("PGBench failed to execute a second time, stopping...")
+                exit(1)
+        data = self._parse_stdout(output[0])
+        progress = self._parse_stderr(output[1])
+        documents = self._json_payload(self.meta_processed, data)
+        documents_raw = self._json_payload_raw(self.meta_processed, data)
+        documents_prog = self._json_payload_prog(self.meta_processed, progress, data)
+        print(output[0])
+        if len(documents) > 0:
+            self._summarize_data(data, self.run[0], self.uuid, self.database, self.pgb_vers)
+        print("\n")
+        print(documents)
+        print("\n")
+        if len(documents) > 0:
+            yield documents, 'summary'
+
+        if len(documents_raw) > 0:
```

Comment:
this needs to be:
```
            for document in documents_raw:
                yield document, 'summary'
```
We've to yield the individual document and not the list.

Reply:
@aakarshg this still accurate^

Reply:
yep L205 dealt with it so we should be good :)

Reply:
what about line 206? ```yield document, 'raw'``` vs ```yield document, 'summary'``` from your comment

Reply:
aaah I see, that was a bad copy paste example on my end, it should be `raw` indeed... L206 already addressess my comment, so this an be resolved.

Reply:
ack. thanks!